c2corg_ui/static/js/feed.js

```diff
@@ -211,6 +408,157 @@ app.FeedController.prototype.createImageUrl = function(filename, suffix) {
+
+      } else if (p === 'labande_global_rating' || p === 'labande_ski_rating') {
+        ratings['labande_rating'] = this.slashSeparatedRating_(data.labande_global_rating, data.labande_ski_rating);
+
+      } else if (p === 'mtb_up_rating' || p === 'mtb_down_rating') {
+        ratings['mtb_rating'] = this.slashSeparatedRating_(data.mtb_down_rating, data.hiking_mtb_exposition);
+
+      } else if (p === 'rock_required_rating' || p === 'rock_free_rating') {
+        ratings['rock_rating'] = data.rock_free_rating;
+
+        if (data.rock_required_rating && data.rock_free_rating) {
+          // [A0] (without bracket) is showed only if equipment_rating = P1 or P1+, and if aid_rating is empty.
+          var A0 = (data.equipment_rating === 'P1' || data.equipment_rating === 'P1+') && !data.aid_rating;
+          ratings['rock_rating'] += '>' + data.rock_required_rating;
+          ratings['rock_rating'] += A0 ? '[A0]' : '';
+        }
+      }
+
+    }
+  }
+  app.constants.fullRatingOrdered.forEach(function(rating) {
+    if (rating in ratings && ratings[rating]) {
+      fullRatings[rating] = ratings[rating];
+    }
+  });
+
+  return fullRatings[Object.keys(fullRatings)[0]] ? fullRatings : null;
+};
+
+
+/**
+ * Show only one of the area types, the first that is available:
+ * 1) range 2) admin limits 3) country
+ * @param {?Array<Object>} areas
+ * @return {string | null}
+ * @export
+ */
+app.FeedController.prototype.showArea = function(areas) {
+
+  if (areas !== undefined) {
+
+    // the areas often come in different orders within 3 area objects.
+    var orderedAreas = {'range': [], 'admin_limits': [], 'country': []};
+    var type;
+
+    if (areas.length > 0) {
+      type = areas[areas.length - 1]['area_type'];
+      orderedAreas[type].push(areas[areas.length - 1]['locales'][0]['title']);
+    }
+
+    for (var t in orderedAreas) {
+      if (orderedAreas[t].length) {
+        return orderedAreas[t][orderedAreas[t].length - 1];
+      }
+    }
+
+  }
+  return null;
+};
+
+
+/**
+ * Show title with the good language:
+ * 1) range 2) admin limits 3) country
+ * @param {?Array<Object>} locales
+ * @return {string | null}
+ * @export
+ */
+app.FeedController.prototype.showTitle = function(locales) {
+
+  for (var i = 0; i < locales.length; i++) {
+    if (locales.lang == this.lang_) {
+      if (!locales[i].title_prefix) {
+        return locales[i].title;
+      } else {
+        return locales[i].title_prefix  + ' : ' + locales[i].title;
+      }
+    }
+  }
+
+  if (!locales[0].title_prefix) {
+    return locales[0].title;
+  } else {
+    return locales[0].title_prefix  + ' : ' + locales[0].title;
+  }
+};
+
+
+/**
+ * Show summaru with the good language:
+ * 1) range 2) admin limits 3) country
+ * @param {?Array<Object>} locales
+ * @return {string | null}
+ * @export
+ */
+app.FeedController.prototype.showSummary = function(locales) {
+
+  for (var i = 0; i < locales.length; i++) {
+    if (locales.lang == this.lang_) {
```

Comment:
Are you sure this works?
Perhaps you mean ``if (locales[i].lang == this.lang_)``?

Anyway I am a bit surprised you have to do this. Is there that kind of code in the cards showing localized contents?